geonode/security/middleware.py

```diff
@@ -169,28 +169,20 @@ def __init__(self, get_response):
     def process_request(self, request):
         whitelist = getattr(settings, 'ADMIN_IP_WHITELIST', [])
         if len(whitelist) > 0:
-            user = None
-            if getattr(request, "user", None):
-                user = request.user
-            else:
-                user = extract_user_from_headers(request)
-            if user and user.is_superuser:
+            potential_admins = []
+            potential_admins.append(extract_user_from_headers(request))
+            if hasattr(request, "user"):
+                potential_admins.append(request.user)
+
+            is_admin = any([u.is_superuser for u in potential_admins])
+
+            if is_admin:
                 visitor_ip = visitor_ip_address(request)
-                in_whitelist = False
-                if visitor_ip:
-                    visitor_ipaddress = ipaddress.ip_address(visitor_ip)
-                    for wip in whitelist:
-                        try:
-                            if visitor_ipaddress in ipaddress.ip_network(wip):
-                                in_whitelist = True
-                                break
-                        except Exception:
-                            pass
-                if not visitor_ip or not in_whitelist:
+                if not is_ipaddress_in_whitelist(visitor_ip, whitelist):
                     try:
-                        if getattr(request, "session", None):
+                        if hasattr(request, "session"):
                             logout(request)
-                        if getattr(request, "user", None):
+                        if hasattr(request, "user"):
                             request.user = get_anonymous_user()
```

Comment:
During the realization of the login `LoginFromApiKeyMiddleware`, we notice that this returns the anonymous guardian user, which is actually a user and not the Django anonymous user who behaves differently.
This is the reason why in the extraction we return the default Django user and not the guardian one:
https://github.com/GeoNode/geonode/blob/6c38de431bae0a5de6878165c38183d89f7acdee/geonode/base/auth.py#L35